src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -338,14 +339,13 @@ void maybeAddPartitions(int minPartitionNum) throws ExecutionException, Interrup
       int partitionDifference = minPartitionNum - partitionNum;
 
       // leader assignments -
-      for (BrokerMetadata brokerMetadata : brokers) {
+      while (newPartitionAssignments.size() != partitionDifference) {
```

Comment:
What is the purpose of the argument `new ArrayList<>()` above in line 338? Looks like dead code to me.

Reply:
will address your concerns in a sep. PR.

Reply:
https://github.com/linkedin/kafka-monitor/pull/264